apps/recorder/widget.js

```diff
@@ -199,6 +200,9 @@
     var settings = loadSettings();
     if (writeInterval) clearInterval(writeInterval);
     writeInterval = undefined;
+    function writeOnGPS() {writeLog(settings.period)}
```

Comment:
unsure if this is the best place to define `writeOnGPS()`.

Reply:
Yes - the `removeListener` won't work because we'll have just created a new `writeOnGPS` when entering the `reload` function. What we'll want is to have it defined outside of `reload`, so it's captured once.

Something a bit like
```diff
+  function writeOnGPS() {writeLog(settings.period)}
+
   // Called by the GPS app to reload settings and decide what to do
   let reload = function() {
     var settings = loadSettings();
     if (writeInterval) clearInterval(writeInterval);
     writeInterval = undefined;
-    function writeOnGPS() {writeLog(settings.period)}
```

Reply:
Thanks!